src/main/java/dinkplugin/LevelNotifier.java

```diff
@@ -79,15 +82,36 @@ public void attemptNotify() {
         plugin.messageHandler.createMessage(plugin.config.levelSendImage(), body);
     }
 
-    public void handleLevelUp(String skill, int level) {
+    public void handleLevelUp(String skill, int level, int xp) {
         if (plugin.isSpeedrunWorld()) return;
-        if (plugin.config.notifyLevel() && checkLevelInterval(level) && currentLevels.get(skill) != null) {
-            if (level == currentLevels.get(skill)) {
-                return;
+
+        int virtualLevel = level < 99 ? level : getLevelForExperience(xp);
+        Integer previousLevel = currentLevels.put(skill, virtualLevel);
+        if (plugin.config.notifyLevel() && checkLevelInterval(virtualLevel) && previousLevel != null) {
+            if (virtualLevel > previousLevel) {
+                levelledSkills.add(skill);
+                sendMessage = true;
             }
-            levelledSkills.add(skill);
-            sendMessage = true;
         }
-        currentLevels.put(skill, level);
+    }
+
+    private static Integer getLevelForExperience(int xp) {
+        return xp > 0 ? LEVEL_BY_EXP_THRESHOLD.floorEntry(xp).getValue() : 1;
```

Comment:
$O(log(n))$ lookup; the alternative is hard-coding the [thresholds](https://oldschool.runescape.wiki/w/Experience#Virtual_levels) in a bunch of `if` statements